language/stdlib/src/lib.rs

```diff
@@ -90,7 +90,7 @@ pub fn build_stdlib() -> BTreeMap<String, CompiledModule> {
                     .expect("stdlib module dependency failed to verify");
                 // Tag each module with its index in the module dependency order. Needed for
                 // when they are deserialized and verified later on.
-                modules.insert(format!("{}_{}", i, name), module);
+                modules.insert(format!("{:0>3}_{}", i, name), module);
```

Comment:
Do we still need these numerical prefixes @tzakian?

Reply:
this is also breaking `test_that_generated_file_are_up_to_date_in_git` @matbd

Reply:
and that's because all file names are different.... I suppose I should just add all the new file name in `stdlib/compiled/stdlib` right?

Reply:
Sadly we do still need the numerical prefixes since the module verification needs to be done in dependency order. 

I'm not sure if this PR is needed though. In particular if you look at how this prefix is used in sorting we don't use it as a string, but parse this numerical prefix as a u64 and sort based on that, see `language/compiled/src/lib.rs` lines 51-55 for where this is done. 

If we do want to keep this change to the numeric prefix, the `sort_by` key code referenced above should be able to be removed.

Reply:
well the problem is that this API is called to fetch the module to publish (e.g. genesis) and the order is wrong.
Now I don't want to go into why it worked and how a different PR that I am doing unveiled the issue, but the bottom line is that if you get just the modules out of the `BTreeMap`, and you try to publish them on a fresh instance of a VM, you **will** fail because dependency order is bogus. 
As an example, if you look under `stdlib/compiled/stdlib/*` `10_AccountLimits.mv` comes before `2_Errors.mv` and if you loop through the `build_stdlib().values()` publishing (which is the only thing you can do) you **will** fail.
Moreover if you look at the comment you will find out that is not what the code was doing.
Hope I am making some sense here.....

Reply:
> In particular if you look at how this prefix is used in sorting we don't use it as a string, but parse this numerical prefix as a u64 and sort based on that, see language/compiled/src/lib.rs lines 51-55 for where this is done.

that is not what all code is doing but it is a good reference (thanks) and we could require all callers to do the sorting that way first (though it seems a bit of a stretch to me, but...)

Reply:
so thinking some more :) why do we have the map in the first place?

Reply:
It might be useful to provide an API for topologically sorting a `Vec<CompiledModule>` based on dependencies instead of encoding the order in the file names. That way, we know how to sort/verify a random bundle of modules regardless of how they were saved.

I ran into this problem + another one in writing the CLI code. The second problem is that if you have the name of a module in hand and want to find its bytecode on disk, the numerical prefixes force you to iterate through the directory instead of just building a path...

Reply:
> the numerical prefixes force you to iterate through the directory instead of just building a path...
that is funny....

Reply:
What if you have more than 1000 modules! :p 
(for real this is fine for now, but it would nice to automagically pad as needed)

Reply:
> It might be useful to provide an API for topologically sorting a Vec<CompiledModule> based on dependencies instead of encoding the order in the file names.

I originally encoded them in file names because I really didn't anticipate people calling this directly in Rust code. 

As far as providing the sorting, we could do `(BTreeMap<String, usize>, Vec<CompiledModule>)` if that makes things easier?

Reply:
I think we are going with this for now and then we'll revisit as needed. 
I also think that we could (and probably should) have 2 API one that returns just a vector of `CompiledModule` (in the order returned by the compiler) and one that gives you the file name too. For what is worth the one returning the file name as well could be as simple as a pair `(CompiledModule, String)`

Reply:
also for @tnowacki this was the reason for the failures I was seeing. Basically the test framework was using the `FRESH_MOVELANG_STDLIB` which returned a vector of modules with an order that was not "dependencies based"